src/loss.py

```diff
@@ -1,2 +1,32 @@
 from abc import ABC
+import numpy as np
 
+
+class Purity_function(ABC):
+    def __init__(self):
+        pass
+
+    def purity_value(self, label_values: np.array, mask) -> float:
+        pass
+
+
+class Information_gaing(Purity_function):
```

Comment:
Same as above - Pascal case for class names. Also typo - the noun is information gain
```suggestion
class InformationGain(Purity_function):
```

Reply:
For information:
ThisIsPacalCase
thisIsCamelCase
this-is-kebab-case
this_is_snake_case